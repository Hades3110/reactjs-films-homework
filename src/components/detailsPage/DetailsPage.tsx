import { useParams } from "react-router"
import MainBanner from "../mainBanner/MainBanner"
import FilmTrailers from "./movieTrailers/MovieTrailers"
import MoreLikeThis from "./moreLikeThis/MoreLikeThis"
import MoviCredits from "./movieCredits/MovieCredits"
import styles from './detailsPage.module.scss'

const DetailsPage = () => {
    const { id } = useParams()

    return (
        <>
            <MainBanner id={id ? parseInt(id) : 0} />
            <div className={styles.filmTrailersAndCredits}>
                <FilmTrailers id={id ? parseInt(id) : 0} />
                <MoviCredits id={id ? parseInt(id) : 0} />
            </div>
            <MoreLikeThis id={id ? parseInt(id) : 0} />
        </>
    )
}

export default DetailsPage
