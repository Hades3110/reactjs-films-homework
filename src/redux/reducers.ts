import { combineReducers } from "redux";
import filmListPageCountReducer from "./filmListCountAndCategories/reducer";
import filmListSortReducer from "./filmListSort/reducer";
import mainBannerReducer from "./mainBanner/reducer";

const rootReducer = combineReducers({
    pageCounter: filmListPageCountReducer,
    sortType: filmListSortReducer,
    mainBanner: mainBannerReducer
})

export default rootReducer
