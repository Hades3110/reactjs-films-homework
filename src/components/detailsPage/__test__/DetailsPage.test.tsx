import { create } from "react-test-renderer";
import DetailsPage from '../DetailsPage';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../../../redux/reducers';

describe('DetailsPage Component', () => {

  const store = createStore(rootReducer)

  it('DetailsPage snapshot test', () => {
    const component = create(<Provider store={store}><DetailsPage /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});