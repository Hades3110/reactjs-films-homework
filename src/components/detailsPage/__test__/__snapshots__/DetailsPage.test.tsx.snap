// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage Component DetailsPage snapshot test 1`] = `
Array [
  <img
    className="spinner"
    src="test-file-stub"
  />,
  <div
    className="filmTrailersAndCredits"
  >
    <div
      className="trailer"
    >
      No trailer
    </div>
    <div
      className="creditList"
    >
      No information
    </div>
  </div>,
  <img
    className="spinner"
    src="test-file-stub"
  />,
]
`;
